### DISCUSSION TITLE
bin/console sulu:document:subscriber:debug not showing Eventlisteners using Attributes

### DISCUSSION THREAD

**Discussion by rhp-zellerd** [sulu/sulu#7569] [Q&A] [answered]
| Q | A
| --- | ---
| Sulu Version | 2.6.3
| PHP Version | 8.2

# Actual Behavior

When you use the Command `bin/console sulu:document:subscriber:debug create` it does not show EventListeners that are done with the new Attribute Syntax `#[AsEventListener(event: ArticleCreatedEvent::class, method: 'doAnything')]` It only shows EventListeners that extend the EventSubscriberInterface.

# Expected Behavior

The Command also shows EventListeners that are done with the new Attribute Syntax and not only EventListeners extending EventSubscriberInterface.

# Steps to Reproduce

Create EventListener with the new Attribute Syntax
Run bin/console sulu:document:subscriber:debug create Command

**Comment by alexander-schranz** [comment, marked as answer]
@rhp-zellerd `#[AsEventListener(event: ArticleCreatedEvent::class, method: 'doAnything')]` is not a document manager event as so is no shown there. 

The document manager use its own event dispatcher requires special registration of the subscriber / listener and only contains events of document manager here:

https://github.com/sulu/sulu/blob/06e994f07339dc4552ce06a0005706f38f6b38d5/src/Sulu/Component/DocumentManager/Events.php#L14-L132

All other events are part of the Symfony EventDispatcher and you can debug them the same way as any other Symfony Events with the Symfony Debug commands:

```
bin/console debug:event-dispatcher
bin/console debug:event-dispatcher <event_name>/<event_class>
```
